src/Profiles.ts

```diff
@@ -143,130 +134,291 @@ export class Profiles {
+                prompt: description
+            };
         }
+        return options;
+    }
 
-        const quickPickOptions: vscode.QuickPickOptions = {
-            placeHolder: localize("createNewConnection.option.prompt.ru.placeholder", "Reject Unauthorized Connections"),
-            ignoreFocusOut: true,
-            canPickMany: false
-        };
-
-        const selectRU = ["True - Reject connections with self-signed certificates",
-            "False - Accept connections with self-signed certificates"];
-
-        const ruOptions = Array.from(selectRU);
-
-        const chosenRU = await vscode.window.showQuickPick(ruOptions, quickPickOptions);
-
-        if (chosenRU === ruOptions[0]) {
-            rejectUnauthorize = true;
-        } else if (chosenRU === ruOptions[1]) {
-            rejectUnauthorize = false;
-        } else {
-            vscode.window.showInformationMessage(localize("createNewConnection.rejectUnauthorize",
-                "Operation Cancelled"));
-            return undefined;
+    public async createNewConnection(profileName: string): Promise<string | undefined> {
+        let profileType: string;
+        let userName: string;
+        let passWord: string;
+        let port: number;
+        let rejectUnauthorize: boolean;
+        let options: vscode.InputBoxOptions;
+        let isTrue: boolean;
+
+        profileType = await this.getProfileType();
+
+        const schema: {} = await this.getSchema(profileType);
+        const schemaArray = Object.keys(schema);
+
+        const schemaValues: any = {};
+        schemaValues.name = profileName;
+
+        // Go through array of schema for input values
+        for (const value of schemaArray) {
+            switch (value) {
+            case "host" :
+                options = {
+                    placeHolder: localize("createNewConnection.option.prompt.host.placeholder", "host.com"),
+                    prompt: localize("createNewConnection.option.prompt.host",
+                    "Enter a z/OS Host in the format 'host.com'."),
+                };
+                let host = await vscode.window.showInputBox(options);
+                if (!host) {
+                    vscode.window.showInformationMessage(localize("createNewConnection.zosmfURL",
+                        "No valid value for z/OS Host. Operation Cancelled"));
+                    return undefined;
+                }
+                try {
+                    if (host.includes(":")) {
+                        if (host.includes("/")) {
+                            const result = this.parseUrl(host);
+                            if (result === undefined) {
+                                return undefined;
+                            } else {
+                            host = result.host;
+                            if (result.port !== null) {
+                                port = result.port;
+                            }}
+                        } else {
```

Comment:
These if/else branches appear to contain essentially the same logic and could be replaced by:
```typescript
if (!host.includes("/")) {
    host = "https://" + host;
}
const result = this.parseUrl(host);
...
```